docs/source/developer/architecture/backend/algorithms/detector_peak_predictor.rst

```diff
@@ -0,0 +1,30 @@
+Algorithm: `DetectorPeakPredictor`
+==================================
+
+Description:
+------------
+This algorithm predicts detector peaks based on the provided detector peak ingredients.
+It calculates the beta terms, full width at half maximum (FWHM), and generates a list
+of `DetectorPeak` objects for each group, optionally purging duplicates.
```

Comment:
It predicts peak locations based on crystallographic information, and predicts peak widths based on resolution estimates of the instrument.